tests/questionpy_server/api/test_models.py

```diff
@@ -1,23 +1,87 @@
 #  This file is part of the QuestionPy Server. (https://questionpy.org)
 #  The QuestionPy Server is free software released under terms of the MIT license. See LICENSE.md.
 #  (c) Technische Universität Berlin, innoCampus <info@isis.tu-berlin.de>
-
+from hashlib import sha256
 from io import BytesIO
+from itertools import pairwise, starmap
+from operator import ge
+from unittest.mock import Mock
 
+import pytest
 from aiohttp import FormData
+from aiohttp.pytest_plugin import AiohttpClient
 from aiohttp.test_utils import TestClient
 from pydantic import TypeAdapter
 
-from questionpy_server.api.models import PackageInfo
+from questionpy_server.api.models import PackageVersionInfo, PackageVersionsInfo
+from questionpy_server.app import QPyServer
+from questionpy_server.collector.local_collector import LocalCollector
+from questionpy_server.utils.manifest import ComparableManifest
 from tests.conftest import PACKAGE
+from tests.test_data.factories import ManifestFactory
+
+
+@pytest.mark.parametrize(
+    "packages",
+    [
+        # No packages.
+        {},
+        # One package.
+        {"ns1": {"0.1.0"}},
+        # Two packages.
+        {"ns1": {"0.1.0"}, "ns2": {"0.1.0"}},
+        # Multiple versions.
+        {"ns1": {"1.0.0", "0.0.1"}, "ns2": {"1.0.0", "0.1.0", "0.0.1"}},
+        # Multiple versions, unsorted.
+        {"ns1": {"0.0.1", "1.0.0"}, "ns2": {"0.1.0", "0.0.1", "1.0.0"}},
+    ],
+)
+async def test_packages(qpy_server: QPyServer, aiohttp_client: AiohttpClient, packages: dict[str, set[str]]) -> None:
+    async def add_package_version(server: QPyServer, manifest: ComparableManifest) -> None:
+        package_hash = sha256((manifest.short_name + manifest.namespace + str(manifest.version)).encode()).hexdigest()
+        await server.package_collection._indexer.register_package(package_hash, manifest, Mock(spec=LocalCollector))
 
+    manifests: dict[str, dict[str, ComparableManifest]] = {}
+    for namespace, versions in packages.items():
+        for version in versions:
+            expected_manifest = ManifestFactory.build(namespace=namespace, short_name=namespace, version=version)
+            manifests.setdefault(namespace, {})[version] = expected_manifest
+            await add_package_version(qpy_server, expected_manifest)
 
-async def test_packages(client: TestClient) -> None:
+    client = await aiohttp_client(qpy_server.web_app)
     res = await client.request("GET", "/packages")
 
+    # Assert that a valid list of PackageVersionsInfo is returned.
     assert res.status == 200
     data = await res.json()
-    TypeAdapter(list[PackageInfo]).validate_python(data)
+    package_versions_infos: list[PackageVersionsInfo] = TypeAdapter(list[PackageVersionsInfo]).validate_python(data)
+
+    expected_package_count = len(packages)
+    assert len(package_versions_infos) == expected_package_count
+
+    if expected_package_count <= 0:
+        return
```

Comment:
Da der Rest für den leeren case ein No-Op ist, können wir uns das if zugunsten der Lesbarkeit sparen
```suggestion
    assert len(package_versions_infos) == len(packages)
```